ext/opentelemetry-ext-zipkin/src/opentelemetry/ext/zipkin/__init__.py

```diff
@@ -0,0 +1,182 @@
+DEFAULT_PROTOCOL = "http"
+ZIPKIN_HEADERS = {"Content-Type": "application/json"}
+
+SPAN_KIND_MAP = {
+    SpanKind.INTERNAL: None,
+    SpanKind.SERVER: "SERVER",
+    SpanKind.CLIENT: "CLIENT",
+    SpanKind.PRODUCER: "PRODUCER",
+    SpanKind.CONSUMER: "CONSUMER",
+}
+
+SUCCESS_STATUS_CODE = (200, 202)
+
+logger = logging.getLogger(__name__)
+
+
+class ZipkinSpanExporter(SpanExporter):
+    """Zipkin span exporter for OpenTelemetry.
+
+    Args:
+        service_name: Service that logged an annotation in a trace.Classifier
+            when query for spans.
+        host_name: The host name of the Zipkin server
+        port: The port of the Zipkin server
+        endpoint: The endpoint of the Zipkin server
+        protocol: The protocol used for the request.
+        ipv4: Primary IPv4 address associated with this connection.
+        ipv6: Primary IPv6 address associated with this connection.
+    """
+
+    def __init__(
+        self,
+        service_name: str,
+        host_name: str = DEFAULT_HOST_NAME,
+        port: int = DEFAULT_PORT,
+        endpoint: str = DEFAULT_ENDPOINT,
+        protocol: str = DEFAULT_PROTOCOL,
+        ipv4: Optional[str] = None,
+        ipv6: Optional[str] = None,
+    ):
+        self.service_name = service_name
+        self.host_name = host_name
+        self.port = port
+        self.endpoint = endpoint
+        self.protocol = protocol
+        self.url = "{}://{}:{}{}".format(
+            self.protocol, self.host_name, self.port, self.endpoint
+        )
+        self.ipv4 = ipv4
+        self.ipv6 = ipv6
+
+    def export(self, spans: Sequence[Span]) -> "SpanExportResult":
+        zipkin_spans = self._translate_to_zipkin(spans)
+        result = requests.post(
+            url=self.url, data=json.dumps(zipkin_spans), headers=ZIPKIN_HEADERS
+        )
+
+        if result.status_code not in SUCCESS_STATUS_CODE:
+            logger.error(
+                "Traces cannot be uploaded; status code: %s, message %s",
+                result.status_code,
+                result.text,
+            )
+            # TODO: should we retry here?
+            return SpanExportResult.FAILED_NOT_RETRYABLE
+        return SpanExportResult.SUCCESS
+
+    def _translate_to_zipkin(self, spans: Sequence[Span]):
+
+        local_endpoint = {
+            "serviceName": self.service_name,
+            "port": self.port,
+        }
+
+        if self.ipv4 is not None:
+            local_endpoint["ipv4"] = self.ipv4
+
+        if self.ipv6 is not None:
+            local_endpoint["ipv6"] = self.ipv6
+
+        zipkin_spans = []
+        for span in spans:
+            ctx = span.get_context()
+            trace_id = ctx.trace_id
+            span_id = ctx.span_id
+
+            # Timestamp in zipkin spans is int of microseconds.
+            start_timestamp_mus = _nsec_to_usec_round(span.start_time)
+            duration_mus = _nsec_to_usec_round(span.end_time - span.start_time)
+
+            zipkin_span = {
```

Comment:
it looks like there's no handling of the "debug" field here, which could be extracted from the boolean flags in the TraceOptions.

Reply:
theoretically we could include debug as a bit in trace_options, or use the sampled bit and consider that debug: https://www.w3.org/TR/trace-context-1/#trace-flags

Reply:
i didn't see anything referencing a debug field in the zipkin API https://zipkin.io/zipkin-api/#/default/post_spans. can you elaborate on this?

Reply:
Documentation is in that page under Models - Span

debug | booleanTrue is a request to store this span even if it overrides sampling policy.This is true when the X-B3-Flags header has a value of 1.
-- | --

Reply:
Easy enough to use the sampling bit.